### PR TITLE
Change logstash notifier to fit pipeline jobs

### PR DESCRIPTION
**Pull request by hkbao**
Changed LogstashNotifier.java and some other files to fit pipeline step call of logstash.
#JENKINS-44712

Signed-off-by: Haikuan Bao <hkbao2008@gmail.com>

### CODE REVIEW

**Comment by hkbao** [comment]
This allows LogstashNotifier to implement SimpleBuildStep so that it can be used as step in a jenkins pipeline script. Since SimpleBuildStep needs override the _perform_ method with _Run_ type parameter, some other java files are also updated to make it work.

**Comment by yueyehm** [comment]
@jakub-bochenski Would you please review the modified pull request?

**Comment by jakub-bochenski** [comment]
@yueyehm I don't see any new changes since 795774f1830dfd7dc2376a569cf6f855a4b84d1b

**Comment by hkbao** [comment]
@jakub-bochenski I've made changes in the latest commit, could you pls help take a look?

File: src/main/java/jenkins/plugins/logstash/persistence/BuildData.java [hunk 1]
```diff
@@ -166,6 +157,47 @@ public BuildData(AbstractBuild<?, ?> build, Date currentTime) {
...
+    } else {
+      buildHost = StringUtils.isBlank(executor.getDisplayName()) ? "master" : executor.getDisplayName();
+    }
+
+    rootProjectName = projectName;
+    rootProjectDisplayName = displayName;
+    rootBuildNum = buildNum;
+
+    try {
+      buildVariables = build.getEnvironment(listener);
```
Comment (jakub-bochenski) [inline src/main/java/jenkins/plugins/logstash/persistence/BuildData.java]: How are sensitive variables to be handled in pipeline jobs?

`getSensitiveBuildVariables()` is only defined in `AbstractBuild`, so how is pipeline handling plugins like credentials binding?

**Comment by jakub-bochenski** [comment]
@hkbao thanks, looks better now -- I have one more question about handling secrets tough...

**Comment by jakub-bochenski** [comment]
Also there is an open PR #18 that seems to also add pipeline support, but it looks much more complex. 

Unfortunately per @jglick "certainly the code added here is not even close to production quality" and I don't know enough about pipeline do bring it there. Maybe it would be better to try and merge that PR?

**Comment by smiskiewicz1** [comment]
would be great to use this plugin in jenkins 2 pipelines. 18 looks more like rewriting the whole logging system.  I'd just be happy with being able to forward logs and use the plugin.

**Comment by jakub-bochenski** [comment]
@smiskiewicz1 I agree some support is better that no support at all. 
However, as I understand it the current implementation would not filter-out secrets and this is a rather serious issue.

**Comment by yueyehm** [comment]
@jakub-bochenski from Jesse's comment here:
https://stackoverflow.com/questions/30916085/how-do-we-access-the-sensitive-variables-in-a-jenkins-plugin-that-is-workflow-co,
seems we should ignore this block?

**Comment by jakub-bochenski** [comment]
@hkbao agree, but let's leave a comment in the code, and I think a release note about this would also be helpful to users

**Comment by jglick** [comment]
#18 is not about “adding Pipeline support”, it is pretty much unrelated.

**Comment by jakub-bochenski** [comment]
Added comment, now just need to figure out how to do a release note for Jenkins plugin

**Comment by hkbao** [comment]
@jakub-bochenski Since @yueyehm has commit new changes to the branch which is a minor feature enhancement about test result details, Can you do a quick review? And when can we merge it to master?

**Comment by jakub-bochenski** [comment]
@hkbao @yueyehm the changes look OK, but c27f291cf19a32dbde75330f76752c39f4bf5698 seems to be unrelated to pipeline -- can we have them as a a separate PR to keep things clean?

> And when can we merge it to master
@vjm mentioned he wants to test this change on his local setup so I was waiting if he reports any issues. Other than that I see no reasons not to merge (at least up to 598bec1)

**Comment by hkbao** [comment]
I've reverted @yueyehm commit and he will make a new PR for the change.

**Comment by jakub-bochenski** [comment]
@hkbao @yueyehm I'm soon going on vacation for next two weeks and I won't be able to push a new release with this changes -- sorry about that, but some other last minute tasks got in the way.

I will try to do a release the first thing when I come back.

**Comment by Dmitry1987** [comment]
Sorry if it's not the right place to ask,
but I didn't find a way for pipeline job ('logstashSend') to not send all console log to elasticsearch.  While in regular job there's a checkbox for this. How can I set that checkbox in pipeline to avoid sending all log? (just need the build info, without console log)
Thanks!

**Comment by jakub-bochenski** [comment]
Email group or a new GitHub issue would be preferable.

Have you tried setting `maxLines` to 0?
If it doesn't work please file an issue

**Comment by Dmitry1987** [comment]
oh, good idea.. thanks @jakub-bochenski . I thought '0' means unlimited. But I can trick it, and set to '1' if 0 will not work. 1 line will be fine.
thank you :)

**Comment by bobowang2017** [comment]
All the logs are stored in a key in redis, and the key does not expire, so it is not convenient to find it, and the value corresponding to the key will become larger and larger.

**Comment by bobowang2017** [comment]
Is this support pipeline?

**Pull request merged**